hack/install.sh

```diff
@@ -95,45 +100,60 @@ do_install() {
 		curl='busybox wget -qO-'
 	fi
 
+	# check to see which repo they are trying to install from
+	repo='main'
+	if [ "https://test.docker.com/" = "$url" ]; then
+		repo='testing'
+	elif [ "https://experimental.docker.com/" = "$url" ]; then
+		repo='experimental'
+	fi
+
 	# perform some very rudimentary platform detection
 	lsb_dist=''
+	dist_version=''
 	if command_exists lsb_release; then
 		lsb_dist="$(lsb_release -si)"
+		dist_version="$(lsb_release --codename | cut -f2)"
 	fi
 	if [ -z "$lsb_dist" ] && [ -r /etc/lsb-release ]; then
 		lsb_dist="$(. /etc/lsb-release && echo "$DISTRIB_ID")"
+		dist_version="$(. /etc/lsb-release && echo "$DISTRIB_CODENAME")"
 	fi
 	if [ -z "$lsb_dist" ] && [ -r /etc/debian_version ]; then
 		lsb_dist='debian'
+		dist_version="$(cat /etc/debian_version | sed 's/\/.*//' | sed 's/\..*//')"
+		if [ "$dist_version" = "8" ]; then
+			dist_version="jessie"
+		elif [ "$dist_version" = "7" ]; then
+			dist_version="wheezy"
+		fi
 	fi
 	if [ -z "$lsb_dist" ] && [ -r /etc/fedora-release ]; then
 		lsb_dist='fedora'
+		dist_version="$(rpm -qa \*-release | cut -d"-" -f3)"
 	fi
-	if [ -z "$lsb_dist" ] && [ -r /etc/centos-release ]; then
-		lsb_dist='centos'
+	if [ -z "$lsb_dist" ]; then
+		if [ -r /etc/centos-release ] || [ -r /etc/redhat-release ]; then
+			lsb_dist='centos'
+			dist_version="$(rpm -qa \*-release | cut -d"-" -f3)"
+		fi
 	fi
 	if [ -z "$lsb_dist" ] && [ -r /etc/os-release ]; then
 		lsb_dist="$(. /etc/os-release && echo "$ID")"
+		dist_version="$(. /etc/os-release && echo "$VERSION_ID")"
 	fi
 
 	lsb_dist="$(echo "$lsb_dist" | tr '[:upper:]' '[:lower:]')"
 	case "$lsb_dist" in
-		amzn|fedora|centos)
-			if [ "$lsb_dist" = 'amzn' ]; then
-				(
-					set -x
-					$sh_c 'sleep 3; yum -y -q install docker'
-				)
-			else
-				(
-					set -x
-					$sh_c 'sleep 3; yum -y -q install docker-io'
-				)
-			fi
+		amzn)
+			(
+			set -x
+			$sh_c 'sleep 3; yum -y -q install docker'
```

Comment:
@jfrazelle just double checking that the name of the package here is still `docker` and not `docker-engine`.